File: LOO 2°Parcial/Tarea 1/PersonJava/src/personjava/Student.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package personjava;

/**
 * 
 * @author Dimitri Medina Novelo
 */
public class Student extends Persona{
        private String nombre2; 
        private int semestre; 
        private double promedio;
        
        public Student(String nombre, String apellido, int edad, int day, int month, int year,String nombre2, int semestre, double promedio){
                super( nombre,  apellido,  edad,  day,  month,  year);
                this.setNombre2(nombre2);
                this.semestre=(semestre);
                this.promedio=(promedio);
        }
        public String getNombre2() { return this.nombre2; }
        public void setNombre2(String nombre2) { this.nombre2 = nombre2; }
        
        public int getSemestre() { return this.semestre; }
        public void setSemestre(int value) { this.semestre = semestre < 0 ? 0 : semestre; }
        
        public double getPromedio() { return this.promedio; }
        public void setPromedio(double value) { this.promedio = promedio < 0 ? 0 : promedio; }
        
        public void imprimir() {
        System.out.println(toString());
        System.out.print("PROGRAMA: " + this.nombre2 + "\nSEMESTRE: " + this.semestre + "\nPROMEDIO GLOBAL: " + this.promedio+"\n");
    }
}

File: LOO 2°Parcial/Tarea 1/Person/Person/Student.cpp
#include "Student.h"
Student::Student(string nombre, string apellido, int edad, int day, int month, int year,string nombre2, int semestre, double promedio) : Persona(nombre,  apellido,  edad,  day,  month,  year) {
	this->nombre2 = nombre2;
	this->semestre = semestre < 0 ? 0 : semestre;
	this->promedio = promedio < 0 ? 0 : promedio;
}
void Student::setNombre(string nombre2) {
	this->nombre2 = nombre2;
}
string Student::getNombre(void) {
	return this->nombre2;
}
void Student::setSemestre(int semestre) {
	this->semestre = semestre < 0 ? 0 : semestre;
}
int Student::getSemestre(void) {
	return this->semestre;
}
void Student::setPromedio(double promedio) {
	this->promedio = promedio < 0 ? 0 : promedio;
}
double Student::getPromedio(void) {
	return this->promedio;
}
void Student::imprimir(void) {
	cout << Student::toString();
	cout << "\nCARRERA: " << this->nombre2 << "\nSEMESTRE: " << this->semestre << "\nPROMEDIO: " << this->promedio;
}